torchdata/nodes/samplers/stop_criteria.py

```diff
@@ -17,8 +17,12 @@ class StopCriteria:
        dataset is seen exactly once. No wraparound or restart will be performed.
 
     3) FIRST_DATASET_EXHAUSTED: Stop when the first dataset is exhausted.
+
+    4) CYCLE_FOREVER: Cycle through the datasets by reinitializing each exhausted source nodes.
+       This is useful when trainer want control over certain number of steps instead of epochs.
```

Comment:
Will this call set_epoch on the underlying datasets?

Reply:
Wondering if we need to do anything special, or can advise users on how to perform that. This otherwise looks good to me